src/app/parcels/batch/BatchParcelDataGrid.tsx

```diff
@@ -180,29 +182,30 @@ const styledBatchGridDisplayColumns: GridColDef<BatchGridDisplayRow[][number]>[]
         return { ...column, headerAlign: "center", align: "center" };
     });
 
-interface BatchParcelDataGridProps {
-    initialTableState: BatchTableDataState;
-}
-
 export const defaultTableState: BatchTableDataState = {
     overrideDataRow: {
         data: null,
     },
     batchDataRows: [
         {
             id: 1,
-            clientId: "1",
+            clientId: null,
             data: null,
         },
     ],
     clientOverrides: [],
     parcelOverrides: [],
 };
 
-const BatchParcelDataGrid: React.FC<BatchParcelDataGridProps> = ({ initialTableState }) => {
-    const [tableState, _] = useReducer(batchParcelsReducer, initialTableState);
+const BatchParcelDataGrid: React.FC = () => {
+    const [tableState, _] = useLocalStorage(batchParcelsReducer, defaultTableState);
+    const [displayRows, setDisplayRows] = useState<BatchGridDisplayRow[]>(
+        tableStateToBatchDisplayRows(defaultTableState)
+    );
 
-    const displayRows: BatchGridDisplayRow[] = tableStateToBatchDisplayRows(tableState);
+    useEffect(() => {
+        setDisplayRows(tableStateToBatchDisplayRows(tableState));
+    }, [tableState]);
```

Comment:
Rather than `useState` and `useEffect`, can we have a `useMemo` here? This implementation will give us an extra re-render, as well as overcomplicating things a bit. You can read a bit more about it here https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state